File: twosum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> a;
        vector<int> b{0, 0};
        for(int i = 0; i < nums.size(); i++){
            a.insert(map<int, int>::value_type(nums[i], i));
        }
        for(int i = 0; i < nums.size(); i++){
            if(a.count(target - nums[i]) > 0 && a[target - nums[i]] != i){
                b[0] = i;
                b[1] = a[target - nums[i]];
                break;
            }
        }
        return b;
    }
};

File: Dict/twosum.py
class solution:
     def twoSum(self, nums, target):
          # visited里面存储数值 -> 下标
          visited = {}
          for i, num in enumerate(nums):
               need = target - num
               if need in visited.keys():
                    return [i, visited[need]]
               visited[num] = i
